functions/src/features/accountManagement.ts

```
@@ -0,0 +1,17 @@
+export const checkWhitelist = onCall(async (req) => {
+  return new Promise(async (resolve, reject) => {
+    // TODO: Check if user email is somewhere in database, delete account if not
+    const uid = await req.auth?.uid;
+    if (!uid) {
+      reject("Unauthenticated");
+      throw new HttpsError("failed-precondition", "Unauthenticated");
+    }
+
+    await adminAuth.setCustomUserClaims(uid, { role: "ADMIN" });
+    resolve(`User with uid ${uid} has been given ADMIN role`);
+  });
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)